### PR TITLE
Unset TL_CONFIG in ContaoTestCase::tearDown()

### PR DESCRIPTION
**Pull request by fritzmg**
Currently the CI fails with many tests that extend from `ContaoTestCase` and use the `mockContaoFramework` method, due to the global state change watcher which detects that `$GLOBALS['TL_CONFIG']` was set.

This is because `mockContaoFramework` will automatically load the default `TL_CONFIG` configuration (and thus `$GLOBALS['TL_CONFIG']` gets populated).

This PR would fix that by unsetting the global in the `tearDown` in this base class.

### CODE REVIEW

**Comment by leofeyer** [comment]
The global state watcher has been added in Contao 4.13, therefore this pull should be rebased.

File: test-case/src/ContaoTestCase.php
Comment (leofeyer) [inline test-case/src/ContaoTestCase.php:58]: Maybe you can also delete the `tearDown()` methods in the tests that do nothing more than this?

**Comment by fritzmg** [comment]
> The global state watcher has been added in Contao 4.13, therefore this pull should be rebased.

Yes, but shouldn't we clean up the global state in 4.9 regardless?

**Comment by Toflar** [comment]
I think @ausi should review this as he's the author of the feature.

**Review by ausi — DISMISSED** [review]

File: test-case/src/ContaoTestCase.php
Comment (fritzmg) [reply test-case/src/ContaoTestCase.php:58]: Yes, will do once this is merged and then merged upstream.

Comment (leofeyer) [reply test-case/src/ContaoTestCase.php:58]: Please rebase this PR to the 4.13 branch and make the changes here.

**Review by m-vo — DISMISSED** [review]

File: test-case/src/ContaoTestCase.php
Comment (fritzmg) [reply test-case/src/ContaoTestCase.php:58]: Done in 6cf9fe27b7224529121dbb6ba0c42b3c6c6cb964

**Review by ausi — DISMISSED** [review]

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @fritzmg.

**Pull request merged**
